UnitTests/Common/Decorators/DispatcherLoggingDecoratorTests.cs

```diff
@@ -0,0 +1,34 @@
+using Xunit;
+using ViaEventAssociation.Core.Application.AppEntry;
+using ViaEventAssociation.Core.Application.AppEntry.Commands; 
+using Microsoft.Extensions.DependencyInjection;
+using ViaEventAssociation.Core.Domain.Aggregates.Events.ValueObjects;
+using UnitTests.Common.Dispatcher;
+using ViaEventAssociation.Core.Application.AppEntry.Common;
+
+namespace UnitTests.Common.Decorators;
+
+public class DispatcherLoggingDecoratorTests
+{
+    [Fact]
+    public async Task DispatchAsync_ShouldLogBeforeAndAfter()
+    {
+        // Arrange
+        IServiceCollection serviceCollection = new ServiceCollection();
+        serviceCollection.AddScoped<ICommandHandler<CreateEventCommand>, CreateEventMockHandler>();
+        IServiceProvider serviceProvider = serviceCollection.BuildServiceProvider();
+        var innerDispatcher = new CommandDispatcher(serviceProvider);
+        var id = EventId.Create();
+        var command = CreateEventCommand.Create(id.Data.Id.ToString());
+        var handler = (CreateEventMockHandler) serviceProvider.GetService<ICommandHandler<CreateEventCommand>>()!;
+
+        var logger = new ConsoleLogger();
+        var decorator = new DispatcherLoggingDecorator(innerDispatcher, logger);
+
+        // Act
+        await decorator.DispatchAsync(command);
+
+        // Assert
```

Comment:
need to check if the right handler called :
Assert.True(handler.Called);
here we verify that the right handler is called